bool foot1
bool foot2
bool foot3
bool foot4
